chromium_src/chrome/browser/ui/views/tabs/z_orderable_tab_container_element.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/views/tabs/z_orderable_tab_container_element.h"
+
+#include "brave/browser/ui/views/tabs/features.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+
+namespace brave {
+float CalculateZValue_ChromiumImpl(views::View* child);
+}
+
+// This will replace ZOrderableTabContainer::CalculateZValue to
+// brave::CalculateZValue_ChromiumImpl, which is definition for a function
+// declared in the brave namespace above.
+#define ZOrderableTabContainerElement brave
+#define CalculateZValue CalculateZValue_ChromiumImpl
+#include "src/chrome/browser/ui/views/tabs/z_orderable_tab_container_element.cc"
+#undef CalculateZValue
+#undef ZOrderableTabContainerElement
+
+// static
+float ZOrderableTabContainerElement::CalculateZValue(views::View* child) {
+  if (views::AsViewClass<TabGroupUnderline>(child)) {
+    if (auto* browser_view = BrowserView::GetBrowserViewForNativeWindow(
```

Comment:
what is the `GetBrowserViewForNativeWindow` check for? This needs some comments explaining.

Reply:
for instance see ```
BrowserFeaturePromoController* BrowserFeaturePromoController::GetForView(
    views::View* view) {
  if (!view)
    return nullptr;
  views::Widget* widget = view->GetWidget();
  if (!widget)
    return nullptr;

  BrowserView* browser_view = BrowserView::GetBrowserViewForNativeWindow(
      widget->GetPrimaryWindowWidget()->GetNativeWindow());
  if (!browser_view)
    return nullptr;

  return browser_view->GetFeaturePromoController();
}
```

Reply:
it appears we are also doing this in other places like tab_drag_controller which should also have null checks

Reply:
This seems to be a good time for me to try remove additional widget for vertical tabstrip.

https://github.com/brave/brave-browser/issues/26891